### PR TITLE
feat: adds question create form

### PR DESCRIPTION
**Pull request by thisislawatts**
PR Checklist

- [X] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Introduces initial flow for generating Questions on the platform. 

Screenshot 2023-10-22 at 21 42 35
Screenshot 2023-10-22 at 21 42 39
Screenshot 2023-10-22 at 21 42 51

### CODE REVIEW

File: src/pages/Question/QuestionListing.tsx [hunk 1]
```diff
@@ -1,9 +1,64 @@
...
+        <Link to={'/questions/create'}>
+          <Button variant={'primary'}>Ask a question</Button>
+        </Link>
+      </Flex>
+      {isLoading ? (
+        <Loader />
+      ) : questionList.length ? (
+        questionList.map((q: any, idx) => (
+          <Card key={idx} mb={3} px={3} py={3}>
+            <Link to={`/question/${encodeURIComponent(q.slug)}`} key={q._id}>
```
Comment (benfurber) [inline src/pages/Question/QuestionListing.tsx:45]: Is this to value complex enough that it's worth pulling out into a const?

File: src/pages/Question/QuestionCreate.tsx [hunk 2]
```diff
@@ -1,9 +1,139 @@
...
+                      The Question
+                    </Label>
+                    <Field
+                      name="title"
+                      id="title"
+                      validate={composeValidators(required, minValue(10))}
+                      component={FieldInput}
+                      placeholder="How come … does not work?"
+                      minLength={10}
+                      maxLength={60}
```
Comment (AlfonsoGhislieri) [inline src/pages/Question/QuestionCreate.tsx:61]: **Suggestion**: Can we pull these numbers into a constants file?

File: src/pages/Question/QuestionCreate.tsx [hunk 3]
```diff
@@ -1,9 +1,139 @@
...
+                      Give some more information
+                    </Label>
+                    <Field
+                      name="description"
+                      id="description"
+                      label="Information"
+                      validate={composeValidators(required)}
+                      component={FieldInput}
+                      placeholder="Introduce to your research question. Mention what you want to do, whats the goal what challenges you see, etc"
+                      maxLength={1000}
```
Comment (AlfonsoGhislieri) [inline src/pages/Question/QuestionCreate.tsx:77]: Same as above

File: src/pages/Question/QuestionEdit.tsx
Comment (AlfonsoGhislieri) [inline src/pages/Question/QuestionEdit.tsx:1]: **Question**: is this just meant to be a scaffold for now?

Comment (thisislawatts) [reply src/pages/Question/QuestionEdit.tsx:1]: Thats right, nothing functional in here as part of this PR. My preference is for smaller PRs that incrementally introduce the feature.

File: src/pages/Question/QuestionEdit.tsx [hunk 4]
```diff
@@ -1,6 +1,27 @@
...
+type IProps = RouteComponentProps<{ slug: string }>
+
+export const QuestionEdit = (props: IProps) => {
+  const store = useQuestionStore()
+  const [isLoading] = useState(true)
+
+  useEffect(() => {
+    const { slug } = props.match.params
+    const fetchQuestions = async () => {
+      const question: any = await store.fetchQuestionBySlug(slug)
```
[the remark this answers is not in the document]
Comment (thisislawatts) [reply src/pages/Question/QuestionEdit.tsx]: Removed in b4e3e2a9e6a5c0be340bf33fd9ec651150ec15fd

File: src/pages/Question/QuestionCreate.tsx [hunk 5]
```diff
@@ -1,9 +1,139 @@
...
+                </Card>
+                <Card mt={3} p={4}>
+                  <Box mb={3}>
+                    <Label htmlFor="title" sx={{ fontSize: 2, mb: 2 }}>
+                      The Question
+                    </Label>
+                    <Field
+                      name="title"
+                      id="title"
+                      validate={composeValidators(required, minValue(10))}
```
[the remark this answers is not in the document]
Comment (thisislawatts) [reply src/pages/Question/QuestionCreate.tsx]: Moved as part of bf75a9c435b86e82539f78ea15a2f2b245ce928f

File: src/pages/Question/QuestionCreate.tsx [hunk 6]
```diff
@@ -1,9 +1,139 @@
...
+  const store = useQuestionStore()
   return (
     <Box sx={{ p: 7 }}>
-      <Heading>Question Create</Heading>
+      <Form
+        onSubmit={async (v: any) => {
+          try {
+            const newDocument = await store.upsertQuestion(v)
+            if (newDocument) {
+              // eslint-disable-next-line no-console
```
[the remark this answers is not in the document]
Comment (thisislawatts) [reply src/pages/Question/QuestionCreate.tsx]: Fixed in bf75a9c435b86e82539f78ea15a2f2b245ce928f

File: src/pages/Question/QuestionCreate.tsx [hunk 7]
```diff
@@ -1,9 +1,139 @@
...
+                  <Box mb={3}>
+                    <Label htmlFor="title" sx={{ fontSize: 2, mb: 2 }}>
+                      The Question
+                    </Label>
+                    <Field
+                      name="title"
+                      id="title"
+                      validate={composeValidators(required, minValue(10))}
+                      component={FieldInput}
+                      placeholder="How come … does not work?"
```
[the remark this answers is not in the document]
Comment (thisislawatts) [reply src/pages/Question/QuestionCreate.tsx:65]: Is this a blocking comment for this PR or something we can pick up in a future iteration? I would like to focus on the behaviour and do a clean up refactor once I have all of the forms configured.

File: src/pages/Question/QuestionListing.tsx (hunk 1 above)
Comment (thisislawatts) [reply src/pages/Question/QuestionListing.tsx]: Addressed in 1743114852accdd690d966497553057ad8d29e6c

**Review by AlfonsoGhislieri — APPROVED** [review]
Thanks for the changes and adding the constants, looks good to me! Onwards and upwards with the Question module! 🎉

**Pull request merged**